libs/scan.py

```diff
@@ -70,159 +73,118 @@ def draw_sg(security_group, sg_raw):
-                value = report[vpc]['Subnets'][subnet]['EC2'][ec2]
-                if ('PrivateIpAddress' in value and record_value == value['PrivateIpAddress']) or \
-                    ('Name' in value and record_value == value['Name']) or \
-                    ('PublicIpAddress' in value and record_value == value['PublicIpAddress']):
-                    report[vpc]['Subnets'][subnet]['EC2'][ec2]['DnsRecord'] = record['Name'].replace('\\052', '*')
-            for elbv2 in report[vpc]['Subnets'][subnet]['ELBV2']:
-                value = report[vpc]['Subnets'][subnet]['ELBV2'][elbv2]
-                if ('DNSName' in value and record_value == f'{value["DNSName"]}.'):
-                    report[vpc]['Subnets'][subnet]['ELBV2'][elbv2]['DnsRecord'] = record['Name'].replace('\\052', '*')
-
-def s3_scan_concat_permissions(s3_report, acls, permission, right, override=False):
-    """
-    Scan ACLS for S3 Buckets and appends the right
-    """
-    map_users_uri = {
-        'http://acs.amazonaws.com/groups/global/AllUsers': 'ACL: All Users',
-        'http://acs.amazonaws.com/groups/global/AuthenticatedUsers': 'ACL: Any Authenticated Users',
-        'http://acs.amazonaws.com/groups/s3/LogDelivery': 'ACL: S3 Log Delivery'
-    }
-    for grant in acls:
-        if isinstance(permission, list) and grant['Permission'] not in permission:
-            continue
-        if isinstance(permission, str) and grant['Permission'] != permission:
-            continue
-        if 'URI' in grant['Grantee'] and grant['Grantee']['URI'] in map_users_uri:
-            if override or map_users_uri[grant['Grantee']['URI']] not in s3_report:
-                s3_report[map_users_uri[grant['Grantee']['URI']]] = right
-            else:
-                s3_report[map_users_uri[grant['Grantee']['URI']]] = f'{s3_report[map_users_uri[grant["Grantee"]["URI"]]]},{right}'
-            if map_users_uri[grant['Grantee']['URI']] in ['ACL: All Users', 'ACL: Any Authenticated Users']:
-                s3_report['PubliclyAccessible'] = True
-    return s3_report
-
-def s3_scan_acls(s3_report, acls):
-    """
-    Scan ACLS for S3 Buckets
-    """
-    s3_report = s3_scan_concat_permissions(s3_report, acls, 'READ', 'LIST')
-    s3_report = s3_scan_concat_permissions(s3_report, acls, 'READ_ACP', 'READ')
-    s3_report = s3_scan_concat_permissions(s3_report, acls, ['WRITE', 'WRITE_ACP'], 'WRITE')
-    s3_report = s3_scan_concat_permissions(
-        s3_report,
-        acls,
-        'FULL_CONTROL',
-        'LIST,READ,WRITE',
-        override=True)
-    return s3_report
+    for i, asset in enumerate(assets):
+        asset_type = asset.get_type()
+        if asset_type == 'EC2' and record_value in (asset.public_ip, asset.private_ip):
+            assets[i].dns_record = record['Name'].replace('\\052', '*')
+        elif asset_type == 'ELBV2' and record_value == f'{asset.name}.':
+            assets[i].dns_record = record['Name'].replace('\\052', '*')
```

Comment:
Why do you need to replace it only here? 
Is it not possible to do:
```python3
assets[i].dns_record = record['Name'].replace('\\052', '*')
```
Except if you want to keep `\\052`, I  don't think you need to check to replace it

Reply:
Yes, it's only here

Reply:
I mean, you want to keep `\\052` in the other ones?